
<nav class="navbar navbar-default">
  <div class="container-fluid">
    <div class="navbar-header" style="float: left">
        <a class="navbar-brand" href="0_Forside.ipynb" target="_self"> <h2> &uarr; Tilbake til forsiden</h2></a>
    </div>
  </div>
</nav>

# Nedbetaling av lån

**Læringsmål:**

* While-løkker med oppsamling

**Introduksjon:**

Annuitetslån er lån der man betaler et like stort beløp i det antall terminer som lånet løper (med mulig unntak av aller siste termin). Ved slike lån går mye av dette terminbeløpet til renter i starten, etter hvert blir en stadig større andel av terminbeløpet avdrag (tilbakebetaling). I tabellen under har vi vist hvordan et annuitetslån på $1\ 000 \text{ kr}$ nedbetales over $3$ terminer med en terminrente på $5 \text{ \%}$ og terminbeløp på $400 \text{ kr}$. Legg merke til at det siste terminbeløpet er redusert, da avdraget ikke skal overskride restgjelden.

| Termin | Terminbeløp | Rente | Avdrag | Restgjeld |
| --: | --: | --: | --: | --: |
| 0 | | | | 1000.0 |
| 1 | 400.0 | 50.0 | 350.0 | 650.0 |
| 2 | 400.0 | 32.5 | 367.5 | 282.5 |
| 3 | 296.6 | 14.1 | 282.5 | 0.0 |

Det som betales i rente i en termin regnes ut med formel 1:
$$\text{rente} = \text{restgjeld} \cdot \frac{\text{terminrente i prosent}}{100} \tag{1}$$
Avdragsbeløpet for hver termin vil da være gitt ved formelen : $\text{avdrag} = \text{terminbeløp} - \text{rente}$.

## a) 
Fullfør funksjonen `registrer_innbetaling` nedenfor. Funksjonen skal ta inn lånebeløp (gjeld), rentesats i prosent og terminbeløp som parametre, og returnere restgjelden som gjenstår etter innbetalingen.

In [2]:
def registrer_innbetaling(gjeld, rentesats, terminbelop):
    # Beregn rente
    rente = gjeld * (rentesats / 100)
    
    # Beregn avdrag
    avdrag = terminbelop - rente
    
    # Oppdater restgjeld
    ny_gjeld = gjeld - avdrag
    
    return ny_gjeld

# Skriv funksjonskall for å sjekke funksjonen nedenfor:
registrer_innbetaling(1000, 5, 400)

650.0

Tips: du kan bruke funksjonskall med parametrene fra eksempelet ovenfor til å sjekke om funksjonen din fungerer slik den skal:
```Python
print(f"{registrer_innbetaling(1000, 5, 400) = }")
print(f"{registrer_innbetaling(650, 5, 400) = }")
```
Funksjonskallene skal gi følgende utskrift:<br>
`registrer_innbetaling(1000, 5, 400) = 650.0`<br>
`registrer_innbetaling(650, 5, 400) = 282.5`

In [ ]:
# Denne cellen brukes under retting. La stå!

## b) 
Bruk en `while`-løkke til å fullføre funksjonen `finn_nedbetalingstid`, som bruker gjentatte kall til `registrer_innbetaling` frem til hele lånet er nedbetalt. Funksjonen skal returnere nedbetalingstiden gitt i antall terminer (hvor mange ganger må terminbeløpet betales inn for at lånet skal være nedbetalt).



In [1]:
def finn_nedbetalingstid(gjeld, rentesats, terminbelop):
    antall_terminer = 0
    while gjeld > 0:
        rente = gjeld * rentesats/100
        terminbelop = min(terminbelop, gjeld+rente)
        gjeld += rente - terminbelop
        antall_terminer += 1
    return antall_terminer

# Skriv funksjonskall for å sjekke funksjonen nedenfor:
print(f"{finn_nedbetalingstid(1000, 5, 400) = }")

finn_nedbetalingstid(1000, 5, 400) = 3


Tips: du kan bruke et funksjonskall med parametrene fra eksempelet i starten av oppgaven til å sjekke om funksjonen din fungerer slik den skal:
```Python
print(f"{finn_nedbetalingstid(1000, 5, 400) = }")
```
Skal gi følgende utskrift:<br>
`finn_nedbetalingstid(1000, 5, 400) = 3`

In [ ]:
# Denne cellen brukes under retting. La stå!

## c)
Fullfør funksjonen `finn_samlet_rente` som returnerer det totale rentebeløpet som er innbetalt over hele låneperioden.

In [8]:
def finn_samlet_rente(gjeld, rentesats, terminbelop):
    samlet_rente = 0
    while gjeld > 0:
        rente = gjeld * (rentesats / 100)
        samlet_rente += rente
        gjeld = registrer_innbetaling(gjeld, rentesats, terminbelop)
    return samlet_rente

# Skriv funksjonskall for å sjekke funksjonen nedenfor:
print(f"{finn_samlet_rente(1000, 5, 400) = }")

finn_samlet_rente(1000, 5, 400) = 96.625


Tips: du kan bruke et funksjonskall med parametrene fra eksempelet i starten av oppgaven til å sjekke om funksjonen din fungerer slik den skal. Eksempel:
```Python
print(f"{finn_samlet_rente(1000, 5, 400) = }")
```
Skal gi følgende utskrift:<br>
`finn_samlet_rente(1000, 5, 400) = 96.625`

In [ ]:
# Denne cellen brukes under retting. La stå!

<br>
<nav class="navbar navbar-default">
        <div class="container-fluid">
            <div class="navbar-header" style="float: left">
                <a class="navbar-brand" href="2_terningkast.ipynb" target="_self">&lt; Forrige side: <i>simulering av terningkast</i></a>
                </div>
            <div class="navbar-header" style="float: right">
                <a class="navbar-brand" href="4_betingede_funksjoner_arrays.ipynb" target="_self">Neste side: <i>elementvise operasjoner med løkker</i> &gt;</a>
            </div>
        </div>
</nav>